Assets/MRTK/Services/InputSystem/NearInteractionTouchableVolume.cs

```diff
@@ -52,9 +52,21 @@ public override float DistanceToTouchable(Vector3 samplePoint, out Vector3 norma
                 // inside object, use vector to centre as normal
                 normal = samplePoint - TouchableCollider.bounds.center;
                 normal.Normalize();
-                // Return value less than zero so that when poke pointer is inside
-                // object, it will not raise a touch up event.
-                return -1;
+
+                // Try to calculate the proper penetration distance, to allow more accurate processing of touchable volumes.
+                // Return value less than zero so that when poke pointer is inside object, it will not raise a touch up event.
+                float rayScale = 1.1f;
+                Vector3 outsidePoint = TouchableCollider.bounds.center + normal * (TouchableCollider.bounds.extents.magnitude * rayScale);
+                if (TouchableCollider.Raycast(new Ray(outsidePoint, -normal), out RaycastHit raycastHit, TouchableCollider.bounds.size.magnitude * rayScale))
```

Comment:
I have some concerns about performing a raycast everything this calculation is done. Is there a way to calculate the distance using the existing normal vector?

Reply:
I tried to search for other ways, but didn't find any. 
TouchableCollider ususally is a `BoxCollider`, but can also be any arbitrary collider including a complex mesh collider. So doing any math based upon that assumption basically resolves to a raycast.
Also keep in mind, this raycast is only executed if we're already touching a volume and only for volumes we're touching. Further it's **NOT** a `Physics.Raycast` but a `Collider.Raycast` which is only considering the given collider. Therefore should be hopefully cheap (especially if its a primitive collider).